kitura-init.js

```diff
@@ -1,32 +1,150 @@
 #!/usr/bin/env node
 
 const chalk = require('chalk');
-const spawn = require('child_process').spawn;
+const spawnSync = require('child_process').spawnSync;
+var os = require('os');
+var fs = require('fs');
 
 let args = process.argv.slice(2);
 
-let options = ['-p', 'yo@1', '-p', 'generator-swiftserver', '--', 'yo', 'swiftserver'];
-
 if (args.length > 0) {
     if (args.indexOf('--help') > -1) {
         printHelp();
         process.exit(0);
     }
+}
+
+let path = require("path");
+let currentDirPath = path.resolve("./");
+let currentDir = path.basename(currentDirPath);
+// Replace spaces with hyphens so Xcode project will build.
+let projName = currentDir.replace(/ /g,"-")
+
+if (projName.charAt(0) === '.') {
+     console.error('Application name cannot start with .: %s')
+   }
+   if (projName.toLowerCase() === 'node_modules') {
+    console.error('Application name cannot be {node_modules}')
+   }
+   if (projName.toLowerCase() === 'favicon.ico') {
+     console.error('Application name cannot be {favicon.ico}')
+}
+
+// Make sure directory is empty.
+checkEmptyDir()
+
+// Clone repo contents into current directory.
+cloneProject()
+
+// Rename project to match current directory
+renameProject()
+
+// If '--skip-build' not specified, then build project.
+if (!(args.includes('--skip-build'))) {
+    buildProject()
+}
+
+function checkEmptyDir() {
```

Comment:
Currently this function only works for the current directory. Either it should be called something like `checkCurrentDirIsEmpty()` or it should accept a path parameter that it checks.